install/kubernetes/Makefile

```
@@ -34,80 +29,37 @@ HELM_DOCS := $(DOCKER_RUN) $(HELM_DOCS_IMAGE)
-		# etcd.image.tag												\
-		$(SED) -i '/repository.*etcd-operator.*/{!b;n;s/tag.*/tag: '$(MANAGED_ETCD_VERSION)'/}' $(CILIUM_VALUES)		\
-		# clustermesh.apiserver.etcd.image.tag										\
-		$(SED) -i '/repository.*etcd$$/{!b;n;s/tag.*/tag: '$(ETCD_VERSION)'/}' $(CILIUM_VALUES)				\
-		# nodeinit.image.tag												\
-		$(SED) -i '/repository.*cilium\/startup-script.*/{!b;n;s/tag.*/tag: '$(NODEINIT_VERSION)'/}' $(CILIUM_VALUES)	\
-		# certgen.image.tag												\
-		$(SED) -i '/repository.*certgen.*/{!b;n;s/tag.*/tag: '$(CERTGEN_VERSION)'/}' $(CILIUM_VALUES);			\
-		$(SED) -i 's/'$(CILIUM_PULLPOLICY_REGEX)'/\1 '$$pull_policy'/' $(CILIUM_VALUES);  					\
-		$(SED) -i 's/useDigest:.*/useDigest: '$$use_digest'/' $(CILIUM_VALUES);  							\
-		# image digests;													\
-		$(SED) -i '/# cilium-digest.*/{!b;n;s/digest.*/digest: "'$(CILIUM_DIGEST)'"/}' $(CILIUM_VALUES); 				\
-		$(SED) -i '/# hubble-relay-digest.*/{!b;n;s/digest.*/digest: "'$(HUBBLE_RELAY_DIGEST)'"/}' $(CILIUM_VALUES);			\
-		$(SED) -i '/# operator-aws-digest.*/{!b;n;s/awsDigest.*/awsDigest: "'$(OPERATOR_AWS_DIGEST)'"/}' $(CILIUM_VALUES);                 \
-		$(SED) -i '/# operator-azure-digest.*/{!b;n;s/azureDigest.*/azureDigest: "'$(OPERATOR_AZURE_DIGEST)'"/}' $(CILIUM_VALUES);         \
-		$(SED) -i '/# operator-alibabacloud-digest.*/{!b;n;s/alibabacloudDigest.*/alibabacloudDigest: "'$(OPERATOR_ALIBABACLOUD_DIGEST)'"/}' $(CILIUM_VALUES);         \
-		$(SED) -i '/# operator-generic-digest.*/{!b;n;s/genericDigest.*/genericDigest: "'$(OPERATOR_GENERIC_DIGEST)'"/}' $(CILIUM_VALUES); \
-		$(SED) -i '/# clustermesh-apiserver-digest.*/{!b;n;s/digest.*/digest: "'$(CLUSTERMESH_APISERVER_DIGEST)'"/}' $(CILIUM_VALUES)
-
-CRDS = $(foreach path,$(patsubst %.yaml,%,$(shell find $(ROOT_DIR)/examples/crds/*/ -type f)),$(shell basename $(path)))
-lint:
+all: update-versions docs lint
+
+update-chart: $(ROOT_DIR)/VERSION # Update chart versions to point to the current version.
+	$(ECHO_GEN)cilium/Chart.yaml
+	$(QUIET)grep -lR -e 'version:' -e 'appVersion:' $(CILIUM_CHARTS)/ \
+		| xargs -L 1 $(SED) -i -e 's/'$(CILIUM_CHART_REGEX)'/\1 '$(VERSION)'/g'
+	$(QUIET)$(SED) -i 's;icon:.*;icon: $(LOGO_BASE_URL)/cilium@$(CILIUM_BRANCH)/$(LOGO_PATH);' $(CHART_FILE)
+
+cilium/values.yaml: $(CILIUM_CHARTS)/values.yaml.tmpl $(ROOT_DIR)/VERSION $(ROOT_DIR)/$(RELATIVE_DIR)/Makefile.digests $(ROOT_DIR)/$(RELATIVE_DIR)/$(MAKEFILE_VALUES)
+	$(ECHO_GEN)$@
+	echo -e "# File generated by $(RELATIVE_DIR)/Makefile; DO NOT EDIT." > $@
+	echo -e "# This file is based on $(RELATIVE_DIR)/cilium/values.yaml.tmpl.\n" >> $@
+	envsubst < $< >> $@
+
+update-versions: update-chart cilium/values.yaml # Update the Helm values file to point to the current version.
+
+ALL_HELM_FILES := $(shell find cilium/ -type f)
+CRD_FILES := $(shell find $(ROOT_DIR)/examples/crds/*/ -type f)
+CRDS := $(foreach path,$(patsubst %.yaml,%,$(CRD_FILES)),$(shell basename $(path)))
+lint: cilium/values.yaml $(CRD_FILES) $(ALL_HELM_FILES)
```

My mistake as I've previously asked to add `cilium/values.yaml` to the `lint` target, but since `lint` is phony it's going to be executed regardless of dependency updates.

```suggestion
CRD_FILES := $(shell find $(ROOT_DIR)/examples/crds/*/ -type f)
CRDS := $(foreach path,$(patsubst %.yaml,%,$(CRD_FILES)),$(shell basename $(path)))
lint:
```

---

I think `lint` still needs to depend on `cilium/values.yaml` to ensure that file has been generated against the latest template changes. I guess we can get rid of the extra find for `ALL_HELM_FILES` but it feels a bit weird to add only a subset of the dependencies into the target dependencies :thinking:

---

I guess it doesn't really matter, by default it'll generate that file and then run the linter. I'll take your suggestion here.